kedro-datasets/test_requirements.txt

```diff
@@ -34,7 +34,7 @@ Pillow~=9.0
 plotly>=4.8.0, <6.0
 pre-commit>=2.9.2, <3.0  # The hook `mypy` requires pre-commit version 2.9.2.
 psutil==5.8.0
-pyarrow>=1.0, <7.0
+pyarrow>=1.0, <9.0
```

Comment:
This will need sign off from the wider team if it affects other pieces of the framework - hopefully not an issue.

Reply:
@noklam @merelcht  do you have a view here? What's the process for this sort of change.

Reply:
Or we should include `10.0.0` at least?